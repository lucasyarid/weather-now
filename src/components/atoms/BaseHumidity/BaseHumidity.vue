<template>
  <div class="BaseHumidity">
    <div class="BaseHumidity__title">
      Humidity
    </div>
    <div class="BaseHumidity__value">
      {{ humidity }}<span class="BaseHumidity__scale">%</span>
    </div>
  </div>
</template>

<script>
export default {
  name: "BaseHumidity",
  props: {
    humidity: {
      type: Number,
      required: true
    }
  }
};
</script>

<style lang="scss" scoped>
.BaseHumidity {
  color: $color-darker;
  text-align: center;
  &__title {
    text-transform: uppercase;
    font-size: 12px;
    margin-bottom: 5px;
    font-weight: 100;
  }
  &__value {
    font-size: 21px;
  }
  &__scale {
    font-size: 12px;
  }
}
</style>
