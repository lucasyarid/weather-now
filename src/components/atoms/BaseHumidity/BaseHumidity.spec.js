import { shallowMount } from "@vue/test-utils";
import BaseHumidity from "./BaseHumidity";

describe("BaseHumidity", () => {
  const WRAPPER = shallowMount(BaseHumidity, {
    propsData: {
      humidity: 70
    }
  });

  test("Componente existe", () => {
    expect(WRAPPER.exists()).toBe(true);
  });

  test("Componente nÃ£o alterado", () => {
    expect(WRAPPER.html()).toMatchSnapshot();
  });
});
