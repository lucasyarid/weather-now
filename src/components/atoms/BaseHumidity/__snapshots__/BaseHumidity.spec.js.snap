// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseHumidity Componente n√£o alterado 1`] = `
"<div class=\\"BaseHumidity\\"><div class=\\"BaseHumidity__title\\">
    Humidity
  </div> <div class=\\"BaseHumidity__value\\">
    70<span class=\\"BaseHumidity__scale\\">%</span></div></div>"
`;
